armi/reactor/reactors.py

```diff
@@ -223,6 +220,98 @@ def setOptionsFromCs(self, cs):
         self._circularRingPitch = cs["circularRingPitch"]
         self._automaticVariableMesh = cs["automaticVariableMesh"]
         self._minMeshSizeRatio = cs["minMeshSizeRatio"]
+        self._setupEffectiveDelayedNeutronFraction(cs)
+
+    def _setupEffectiveDelayedNeutronFraction(self, cs):
+        """Process the settings for the delayed neutron fraction and precursor decay constants."""
+        # Verify and set the core beta parameters based on the user-supplied settings
+
+        def _betaValidator(value):
+            """
+            Validate the input for beta.
+            
+            Notes
+            -----
+            Beta can either be provided as a float or list of floats.
+            If the list has a single value or is a float, this is 
+            interpreted as the total delayed neutron fraction. Otherwise,
+            the value is interpreted as the group-wise delayed neutron
+            fractions.
+            """
+            if value is None:
+                return None
+
+            if isinstance(value, list):
+                return [float(val) for val in value]
+            elif isinstance(value, float):
+                return value
+            else:
+                raise ValueError(
+                    "`beta` must be set as a float or a list of floats. "
+                    f"Attempted to assign {value}."
+                )
```

Comment:
This sort of check should be part of the Setting schema. By the time any code gets access to `cs["beta"]` it should already be in its canonical form. Is the voluptuous schema below not performing the conversion?

Reply:
Yeah I think I put the voluptuous checks in after the fact, so I might just be able to delete this now. I'll try it.

Reply:
Yeah this was just redundant with voluptuous so it's gone now.